cli/cmd/install.go

```diff
@@ -0,0 +1,90 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/apex/log"
+	"github.com/spf13/cobra"
+	"github.com/tarantool/tt/cli/cmdcontext"
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/configure"
+	"github.com/tarantool/tt/cli/install"
+	"github.com/tarantool/tt/cli/modules"
+)
+
+var (
+	Reinstall bool
+	Force     bool
+	Verbose   bool
+	Noclean   bool
+	Local     bool
+)
+
+func getInstallFlags() install.InstallationFlag {
+	return install.InstallationFlag{
+		Reinstall: Reinstall,
+		Force:     Force,
+		Verbose:   Verbose,
+		Noclean:   Noclean,
+		Local:     Local,
+	}
+}
+
+// NewInstallCmd creates install command.
+func NewInstallCmd() *cobra.Command {
+	var installCmd = &cobra.Command{
+		Use:   "install [OPTIONS] what",
+		Short: "Install tarantool/tt",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := modules.RunCmd(&cmdCtx, cmd.Name(), &modulesInfo, internalInstallModule, args)
+			if err != nil {
+				log.Fatalf(err.Error())
+			}
+		},
+	}
+	installCmd.Flags().BoolVarP(&Verbose, "verbose", "V", false, "print log to stderr")
+	installCmd.Flags().BoolVarP(&Force, "force", "f", false, "force requirements errors")
+	installCmd.Flags().BoolVarP(&Noclean, "no-clean-tmp", "", false,
+		"don't remove tmp files")
+	installCmd.Flags().BoolVarP(&Reinstall, "reinstall", "", false, "reinstall program")
+	installCmd.Flags().BoolVarP(&Local, "local", "", false,
+		"install from local files")
+	return installCmd
+}
+
+// internalInstallModule is a default install module.
+func internalInstallModule(cmdCtx *cmdcontext.CmdCtx, args []string) error {
+	cliOpts, err := configure.GetCliOpts(cmdCtx.Cli.ConfigPath)
+	if err != nil {
+		return err
+	}
+	flags := getInstallFlags()
+	if _, err := os.Stat(cmdCtx.Cli.ConfigPath); os.IsNotExist(err) {
+		return fmt.Errorf("There is no tarantool.yaml found, please create one")
+	}
+	if cliOpts.App == nil {
+		cliOpts.App = &config.AppOpts{BinDir: filepath.Join(filepath.Dir(cmdCtx.Cli.ConfigPath),
```

Comment:
I consider cliOpts like something that should not be changed or updated by command implementation. If you need some specific path - use separate structure for your purpose.
Suppose, if I want to add some code, that works with cliOpts.App.BinDir, I expect it to have the same value for all the code in tt after calling configure.GetCliOpts(<some_config_path>). So, please move this all defaults init code to cli opts initialization. So all commands will see the same paths.

Reply:
Moved all logic to configure